Spigot-Server-Patches/0531-Update-itemstack-legacy-name-and-lore.patch

```diff
@@ -0,0 +1,91 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: William Blake Galbreath <Blake.Galbreath@GMail.com>
+Date: Wed, 1 Jul 2020 11:57:40 -0500
+Subject: [PATCH] Update itemstack legacy name and lore
+
+
+diff --git a/src/main/java/net/minecraft/server/ItemStack.java b/src/main/java/net/minecraft/server/ItemStack.java
+index 00725529793d6c2fe26ecacf900355b40e31dad8..53dc47ffb48cce2ef160d1d61766e6700f67c6e3 100644
+--- a/src/main/java/net/minecraft/server/ItemStack.java
++++ b/src/main/java/net/minecraft/server/ItemStack.java
+@@ -83,6 +83,52 @@ public final class ItemStack {
+             list.sort((Comparator<? super NBTBase>) enchantSorter); // Paper
+         } catch (Exception ignored) {}
+     }
++
++    private void processText() {
++        NBTTagCompound display = getSubTag("display");
++        if (display != null) {
++            if (display.hasKeyOfType("Name", 8)) {
++                String json = display.getString("Name");
++                if (json != null && json.contains("\u00A7")) {
++                    try {
++                        display.set("Name", convert(json));
++                    } catch (JsonParseException jsonparseexception) {
++                        display.remove("Name");
++                    }
++                }
++            }
++            if (display.hasKeyOfType("Lore", 9)) {
++                NBTTagList list = display.getList("Lore", 8);
++                boolean legacy = false;
++                for (int index = 0; index < list.size(); index++) {
++                    String json = list.getString(index);
++                    if (json != null && json.contains("\u00A7")) {
++                        legacy = true;
++                        break;
++                    }
++                }
++                if (legacy) {
++                    NBTTagList lore = new NBTTagList();
++                    for (int index = 0; index < list.size(); index++) {
++                        String json = list.getString(index);
++                        try {
++                            lore.add(convert(json));
++                        } catch (JsonParseException ignore) {
++                        }
++                    }
++                    display.set("Lore", lore);
++                }
++            }
++        }
++    }
++
++    private NBTTagString convert(String json) {
++        IChatBaseComponent component = IChatBaseComponent.ChatSerializer.jsonToComponent(json);
++        if (component != null) {
```

Comment:
This should validate that the components has a colored text and that italic is null before converting to avoid converting every single load

Reply:
I'm checking for the legacy color char `\u00A7` before running this `convert` method to prevent converting everything

Reply:
@aikar he;s checking for the legacy color char \u00A7 before running this convert method to prevent converting everything